cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-rococo/src/tests/snowbridge.rs

```diff
@@ -442,30 +453,44 @@ fn send_weth_asset_from_asset_hub_to_ethereum() {
 				],
 			)),
 			fun: Fungible(WETH_AMOUNT),
-		}];
-		let multi_assets = VersionedAssets::V4(Assets::from(assets));
+		};
+		let assets = vec![weth_fee_asset.clone(), weth_asset.clone()];
```

Comment:
The benefit of using weth as fee here is make it dynamic rather than a static config on AH as delivery fee.

More changes required for runtime configs.

Reply:
In the new fee design I'm working on (which I've posted about on Slack), for the BridgeHub->Ethereum direction, relayers are going to be rewarded on the source (BridgeHub) after they've relayed proof-of-delivery.

Keep in mind that since BridgeHub only supports DOT, the fee token needs to DOT.

Reply:
That said, you can include an WETH->DOT swap instruction in the XCM if you want :)

Reply:
Yeah, we can add the [ExchangeAsset](https://github.com/Snowfork/polkadot-sdk/blob/f7054e73a0069bd59228a79b23c2cb21df6c4e71/polkadot/xcm/xcm-executor/src/lib.rs#L1184) instruction for that.

Agreee **relayers should be rewarded on the source side, not the destination**.  But does the source here means AH?

I'd prefer we lock the fee asset on AH with the [LockAsset](https://github.com/Snowfork/polkadot-sdk/blob/f7054e73a0069bd59228a79b23c2cb21df6c4e71/polkadot/xcm/xcm-executor/src/lib.rs#L1128) instruction, when proof of delivery received on BridgeHub we send the [UnLockAsset](https://github.com/Snowfork/polkadot-sdk/blob/f7054e73a0069bd59228a79b23c2cb21df6c4e71/polkadot/xcm/xcm-executor/src/lib.rs#L1150) to AH and closing the cost there.

WDYT?

Reply:
I thought about that but it won't work, we need to send the fee to BH and lock it there

Reply:
I mean https://github.com/Snowfork/polkadot-sdk/blob/058a948d80aacece19f248c031edb47b38007850/cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-rococo/src/tests/snowbridge.rs#L472 to lock the fee asset on AH, so you think that won't work?

Reply:
Okey, as discussed I'll revert this change and make things easy to handle fee stuff on BH instead.